src/Orleans/Core/GrainFactory.cs

```diff
@@ -158,16 +158,16 @@ public TGrainInterface GetGrain<TGrainInterface>(long primaryKey, string keyExte
         public Task<TGrainObserverInterface> CreateObjectReference<TGrainObserverInterface>(IGrainObserver obj)
```

Comment:
You can kept that public method async, and that was the one I was worried about backward compat. The other method is internal, so no issues.